src/deepsparse/transformers/server/utils.py

```diff
@@ -0,0 +1,164 @@
+
+    :return: APIConfig instance for server config
+    """
+    parser = argparse.ArgumentParser("Run DeepSparse Inference Server")
+
+    parser.add_argument(
+        "--host",
+        "-H",
+        type=str,
+        default="0.0.0.0",
+        help="The IP address of the hosted model",
+    )
+    parser.add_argument(
+        "-p",
+        "--port",
+        type=int,
+        default="5543",
+        help="The port that the model is hosted on",
+    )
+
+    parser.add_argument(
+        "-w",
+        "--workers",
+        type=int,
+        default=1,
+        help="The number of workers to use for uvicorn",
+    )
+
+    _args = parser.parse_args()
+    return APIConfig(**vars(_args))
+
+
+################################################################################
+### PipelineConfig
+################################################################################
+
+DEFAULT_BATCH_SIZE = 1
+DEFAULT_MAX_LENGTH = 128
+PipelineConfigType = TypeVar(
+    'PipelineEngineSettingsType',
+    bound='PipelineEngineSettings',
+)
+
+
+class PipelineConfig(BaseSettings):
+    """
+    Settings for pipeline should be set via .env file
+    using the following steps:
+    ```
+    set -a
+    export <file>.env
+    set +a
+    ```
+
+    :param model_file_or_stub: path to (ONNX) model file or SparseZoo stub
+    :param task: name of the task to define which pipeline to create. Defaults
+        to "question-answering"
+    :param num_cores: number of CPU cores to run engine with. Default is the
+        maximum available
+    :param batch_size: The batch size to use for pipeline. Defaults to 1.
+        Note: "question-answering" only supports batch_size 1
+    :param max_length: maximum sequence length of model inputs. Default is 128
+    :param scheduler: The scheduler to use for the engine. Can be None, single
+        or multi
+    :param concurrent_engine_requests: Number of concurrent workers to use
+        for sending requests to the Pipeline. Defaults to 3.
+    """
+    model_file_or_stub: str = 'zoo:nlp/question_answering/bert-base/' \
+                              'pytorch/huggingface/squad/' \
+                              'pruned_quant_3layers-aggressive_89'
+    task: str = 'question-answering'
+    num_cores: Optional[int] = None
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE
+    max_length: int = DEFAULT_MAX_LENGTH
+    scheduler: str = 'multi'
+    concurrent_engine_requests: int = 3
+
+    @staticmethod
+    @lru_cache()
+    def get_config() -> PipelineConfigType:
+        """
+        :return: PipelineEngineSettings for setting up the pipeline object
+        """
+        return PipelineConfig()
+
+
+
+################################################################################
+### Utility functions
+################################################################################
+
+def fix_numpy_types(func):
+    """
+    Decorator to fix numpy types in Dicts, List[Dicts], List[List[Dicts]]
```

Comment:
let's update this comment to more specifically say what is going on here beyond `fix` and why this is needed

Reply:
Done!